src/components/creator/schema.ts

```diff
@@ -0,0 +1,282 @@
+    ],
+    nextStep: ({ values }: { values: AnyObject }) => {
+      if (index + 1 < (values[NAME_TASK_TITLES]?.length ?? 0)) {
+        return taskStepName(index + 1);
+      }
+
+      return STEP_DOWNLOAD;
+    },
+  };
+}
+
+export function makeSchema(chrome: ChromeAPI): Schema {
+  const bundles = chrome.getAvailableBundles();
+
+  const taskSteps = [];
+
+  for (let i = 0; i < MAX_TASKS; ++i) {
+    taskSteps.push(makeTaskStep(i));
+  }
+
+  const schema = {
+    fields: [
+      {
+        component: componentTypes.WIZARD,
+        name: 'wizard-learning-resource',
+        isDynamic: true,
+        crossroads: [NAME_TYPE, NAME_TASK_TITLES],
+        fields: [
+          {
+            name: 'step-type',
+            title: 'Select content type',
+            fields: [
+              {
+                component: componentTypes.SELECT,
+                name: NAME_TYPE,
+                label: 'Type',
+                simpleValue: true,
+                options: Object.entries(itemKindMeta).map(([name, value]) => ({
+                  value: name,
+                  label: value.displayName,
+                })),
+                isRequired: true,
+                validate: [REQUIRED],
+              },
+            ],
+            nextStep: {
+              when: 'type',
+              stepMapper: Object.fromEntries(
+                ALL_ITEM_KINDS.map((kind) => [kind, detailsStepName(kind)])
+              ),
+            },
+          },
+          ...ALL_ITEM_KINDS.map((kind) => makeDetailsStep(kind, bundles)),
+          {
+            name: STEP_PANEL_OVERVIEW,
+            title: 'Panel overview',
+            fields: [
+              {
+                component: componentTypes.TEXTAREA,
+                name: NAME_PANEL_INTRODUCTION,
+                label: 'Introduction (Markdown)',
+                resizeOrientation: 'vertical',
+              },
+              {
+                component: componentTypes.FIELD_ARRAY,
+                name: NAME_PREREQUISITES,
+                label: 'Prerequisites',
+                noItemsMessage: 'No prerequisites have been added.',
+                fields: [
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    label: 'Prerequisite',
+                  },
+                ],
+              },
+              {
+                component: componentTypes.FIELD_ARRAY,
+                name: NAME_TASK_TITLES,
+                label: 'Tasks',
+                minItems: 1,
+                maxItems: MAX_TASKS,
+                noItemsMessage: 'No tasks have been added.',
+                initialValue: [''],
+                fields: [
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    label: 'Title',
+                  },
+                ],
+              },
+            ],
+            nextStep: taskStepName(0),
+          },
+          ...taskSteps,
+          {
+            name: STEP_DOWNLOAD,
```

Comment:
The last step has still a submit button. We don't have any submitted action. You can implement custom buttons per step: https://www.data-driven-forms.org/provided-mappers/wizard?mapper=pf4#heading-buttons#buttons

![screenshot-stage foo redhat com_1337-2024 07 10-09_19_20](https://github.com/RedHatInsights/learning-resources/assets/22619452/de06d94f-65c6-4e89-af20-8ea8bc554597)